runtime/src/util/events.rs

```diff
@@ -0,0 +1,127 @@
+use crate::cbor::serialize_vec;
+use crate::ActorError;
+use fvm_shared::event::{ActorEvent, Entry, Flags};
+use serde::ser;
+
+// Codec identifier for CBOR-encoded data.
+const IPLD_CBOR: u64 = 0x51;
+
+const EVENT_TYPE_KEY: &str = "$type";
+
+/// Builder for ActorEvent objects, accumulating key/value pairs.
+pub struct EventBuilder {
+    entries: Result<Vec<Entry>, ActorError>,
+}
+
+impl EventBuilder {
+    /// Creates a new builder with no values.
+    pub fn new() -> Self {
+        Self { entries: Ok(Vec::new()) }
+    }
+
+    /// Initialise the "type" of the event i.e. Actor event type.
+    pub fn event_type(self, _type: &str) -> Self {
```

Comment:
Nit: `event_type` redundantly says "event" - this context will almost always be obvious because it'll be a chained method call from `EventBuilder::new().event_type()`. So I suggest just `type` (or `typ` if that's a keyword).

Reply:
This is done.